src/StoreApi/Routes/CartUpdateCustomer.php

```diff
@@ -135,24 +137,17 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 	}
 
 	/**
-	 * Validate address fields and return the formatted results.
+	 * Format provided address fields.
 	 *
 	 * @throws RouteException Thrown on error.
 	 * @param array $address Address fields.
 	 * @param array $allowed_countries Countries that must be used.
 	 * @return array
 	 */
 	protected function prepare_address_fields( $address, $allowed_countries ) {
+		// Addresses require a country, otherwise are not valid--return an empty address.
 		if ( empty( $address['country'] ) ) {
-			throw new RouteException(
-				'woocommerce_rest_cart_invalid_country',
-				sprintf(
-					/* translators: 1: valid country codes */
-					__( 'Address country is missing. Please provide one of the following: %s', 'woo-gutenberg-products-block' ),
-					implode( ', ', array_keys( $allowed_countries ) )
-				),
-				400
-			);
+			return [];
```

Comment:
Can you provide some more explanation on why this was changed (I see the phpdoc but it's still not clear to me why this was changed from an exception to returning an empty array).

Reply:
Express payments don't always return a country code I found in testing. Additionally (and similarly) sometimes a billing email is provided but not the full address. This resulted in error 400.

At cart level, having a full complete address is not that important. We just want to track as much data as possible up until an order is created. This matches session behaviour in core. 

I'll update the doc block.